msticpy/data/drivers/sentinel_query_reader.py

```diff
@@ -0,0 +1,452 @@
+    parsed_query_dict = {}
+
+    for query in yaml_queries:
+        with open(query, encoding="utf8", errors="ignore") as opened_query_file:
+            parsed_query_dict[query] = opened_query_file.read()
+
+    return parsed_query_dict
+
+
+def import_sentinel_queries(yaml_files: dict, query_type: str) -> list:
+    """
+    Create list of SentinelQuery attr objects.
+
+    Parameters
+    ----------
+    yaml_files : dict
+        Dictionary mapping query file addresses to yaml file text created by read_yaml_files
+    query_type : str
+        Either "Hunting Queries" or "Detections" or otherwise named query category
+
+    Returns
+    -------
+    list
+        Returns a list of SentinelQuery attr objects from a dict of yaml files and query type given
+
+    """
+    return [
+        _import_sentinel_query(yaml_path, yaml_text, query_type)
+        for yaml_path, yaml_text in yaml_files.items()
+    ]
+
+
+def _import_sentinel_query(
+    yaml_path: str, yaml_text: str, query_type: str
+) -> SentinelQuery:
+    """
+    Create a SentinelQuery attr object for a given yaml query.
+
+    Parameters
+    ----------
+    yaml_path : str
+        File path to a YAML Sentinel query
+    yaml_text : str
+        YAML text (Sentinel query) associated with the yaml_path
+    query_type : str
+        Either "Hunting Queries" or "Detections" or otherwise named query category
+
+    Returns
+    -------
+    SentinelQuery
+        Returns an attrs object called SentinelQuery with all the YAML query information
+
+    """
+    try:
+        parsed_yaml_dict = yaml.load(yaml_text, Loader=yaml.SafeLoader)
+        new_query = SentinelQuery(
+            name=parsed_yaml_dict.get("name"),
+            query_id=parsed_yaml_dict.get("id", ""),
+            description=parsed_yaml_dict.get("description", ""),
+            severity=parsed_yaml_dict.get("severity", ""),
+            tags=parsed_yaml_dict.get("tags_entry", []),
+            required_data_connectors=parsed_yaml_dict.get("requiredDataConnectors", {}),
+            query_frequency=parsed_yaml_dict.get("queryFrequency", ""),
+            query_period=parsed_yaml_dict.get("queryPeriod", ""),
+            trigger_operator=parsed_yaml_dict.get("triggerOperator", ""),
+            trigger_threshold=parsed_yaml_dict.get("triggerThreshold", ""),
+            tactics=parsed_yaml_dict.get("tactics", ""),
+            relevant_techniques=parsed_yaml_dict.get("relevantTechniques", []),
+            query=parsed_yaml_dict.get("query", ""),
+            version=parsed_yaml_dict.get("version", ""),
+            kind=parsed_yaml_dict.get("kind", ""),
+            folder_name=yaml_path.replace("\\", "/").split("/")[-2],
+            source_file_name=yaml_path,
+            query_type=query_type,
+        )
+        if new_query is None:
+            print("No info for query - query is None")
+            print(yaml_path)
+            print(yaml_text)
+        return new_query
+
+    except Exception as error:
```

Comment:
In the except blocks for the previous section, would be good places to put
logger.warning("Failed to parse yaml for %s", yaml_path)

logger.warning("Format error when parsing %s", yaml_path)